prog.go

```diff
@@ -256,6 +256,10 @@ func newProgramWithOptions(spec *ProgramSpec, opts ProgramOptions, handles *hand
 		return nil, fmt.Errorf("can't load %s program on %s", spec.ByteOrder, internal.NativeEndian)
 	}
 
+	if err := fixupAndValidate(spec.Instructions); err != nil {
```

Comment:
This modifies spec.Instruction instead of operating on a copy. It's a bit opaque, currently the copy happens in core.Apply. This changes with the metadata PR.

Reply:
I've respun this for the check to take place if program creation fails, this should no longer be an issue.